### ISSUE TITLE
aws_flow_log - add support for sending to S3

### ISSUE THREAD

**Issue by ryandeivert** [terraform-providers/terraform-provider-aws#5482]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Support S3 as a delivery destination for VPC Flow Logs. As of 2018-08-08, AWS announced support for [sending vpc flow logs to s3](https://aws.amazon.com/about-aws/whats-new/2018/08/amazon-vpc-flow-logs-can-now-be-delivered-to-s3/) (as oppose to the currently supported CloudWatch Logs group). The [API docs](https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_CreateFlowLogs.html) are updated to include this support, and the [`aws-sdk-go`](https://github.com/aws/aws-sdk-go/pull/2091/) has been updated to support this. The interface has changed a bit for this API, and different parameters are now used.

### New or Affected Resource(s)

* aws_flow_log

### Potential Terraform Configuration

```hcl
resource "aws_flow_log" "test_flow_log" {
  log_destination_type = "s3"
  log_destination      = "${aws_s3_bucket.test_bucket.arn}"
  iam_role_arn         = "${aws_iam_role.test_role.arn}"
  resource_type        = "VPC"
  resource_ids         = ["${aws_vpc.default.id}"]
  traffic_type         = "ALL"
}
```

### References

* N/A

**Comment by bflad** [comment]
Support for new `log_destination` and `log_destination_type` arguments to enable sending to S3 has been merged and will release with version 1.42.0 of the AWS provider sometime in the next week or so. Thanks @ryandeivert and @ewbankkit!

**Comment by bflad** [comment]
This has been released in [version 1.42.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1420-october-31-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
